dotcom-rendering/src/web/components/Figure.tsx

```diff
@@ -189,6 +194,8 @@ export const Figure = ({
 			id={id}
 			css={defaultRoleStyles(role)}
 			data-spacefinder-component={spacefinderComponent}
+			data-spacefinder-role={role}
+			data-spacefinder-type={type}
```

Comment:
Having a test of "a figure has `data-spacefinder` attributes from CAPI" would be useful in not regressing in a place where we have dependencies so monitoring errors is hard - and also start to define the contract.

Reply:
Saying this - is the reason for having both so spacefinder can make more specific matches?

Reply:
These attributes are 100% for spacefinder, yeah

Reply:
I have the same question - if spacefinder knows an element has `data-spacefinder-role="immersive"` is this enough information or do we need to be more specific?

Reply:
Sorry, it wasn't very clear! My question is why do we need `data-spacefinder-type` at all? It seems like `data-spacefinder-role` alone is sufficient for the spacefinder rule we need.

Reply:
What @zekehuntergreen said.

Reply:
Gotcha, thanks for clarifying!

So in my head I see `data-spacefinder-type` as a replacement for `data-spacefinder-component`. This latter attribute is - it seems - being used to act differently when the type of element is either a rich link or a numbered list title. Both of these two things are also 'type's so by using 'type' we replace a sort of custom string interface with the actual structured type value of an element.

Note. This type value is only a Frontend/DCR thing and it's not well document or exposed but I think it should have some more limelight

Reply:
Ok got it - thanks for explaining that!

Reply:
Going to broken record this, but having these use cases encapsulated in a test would be great to explain this to our future selves. Maybe something for the commercial team?

Reply:
All that is to say, I'd rather not block this PR with unit tests and would prefer to have the work to define the commercial contract done separately

Reply:
> not block this PR

:shipit: 

>  If the calling code provides a valid prop then it is set as an attribute

This is the test I think makes sense as it is simple and should always pass and validates the `CAPI => DCR => Commercial` pipeline. 

> if we consider an integration test for this

Might be worth widening the discussion to see if we have other examples where this makes sense?

Reply:
> Might be worth widening the discussion to see if we have other examples where this makes sense?

👍 I'm going to add a ticket to review these areas and create the tests